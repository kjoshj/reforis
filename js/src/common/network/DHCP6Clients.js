/*
 * Copyright (C) 2019-2024 CZ.NIC z.s.p.o. (https://www.nic.cz/)
 *
 * This is free software, licensed under the GNU General Public License v3.
 * See /LICENSE for more information.
 */

import React from "react";

import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { formFieldsSize, RichTable } from "foris";
import moment from "moment";
import PropTypes from "prop-types";

DHCP6Clients.propTypes = {
    ipv6clients: PropTypes.arrayOf(PropTypes.object),
};

DHCP6Clients.defaultProps = {
    ipv6clients: [],
};

export default function DHCP6Clients({ ipv6clients }) {
    function renderActiveIcon({ getValue }) {
        return <ActiveIcon isActive={getValue()} />;
    }

    const columns = [
        {
            accessorKey: "hostname",
            header: _("Hostname"),
        },
        {
            accessorKey: "ipv6",
            header: _("IPv6 Address"),
        },
        {
            accessorKey: "duid",
            header: _("DUID"),
        },
        {
            accessorKey: "expires",
            header: _("Expires"),
            cell: ({ getValue }) => {
                const expires = getValue();
                return moment
                    .utc(expires)
                    .locale(ForisTranslations.locale)
                    .format("l LT");
            },
        },
        {
            accessorKey: "active",
            header: _("Active"),
            headerClassName: "d-flex justify-content-center",
            className: "text-center",
            cell: renderActiveIcon,
        },
    ];

    const data = ipv6clients.map((client) => ({
        hostname: client.hostname,
        ipv6: client.ipv6,
        duid: client.duid,
        active: client.active,
        expires: client.expires
            ? moment
                  .unix(client.expires)
                  .locale(ForisTranslations.locale)
                  .format("l LT")
            : _("Never"),
    }));

    return (
        <div className={formFieldsSize}>
            <h2>{_("IPv6 DHCP Client List")}</h2>
            <p>
                {_(
                    "This list contains all devices that are connected to the network through wired or wireless connections using IPv6."
                )}
            </p>
            {ipv6clients.length === 0 ? (
                <p className="text-muted text-center">
                    {_("No clients found.")}
                </p>
            ) : (
                <RichTable columns={columns} data={data} withPagination />
            )}
        </div>
    );
}

ActiveIcon.propTypes = {
    isActive: PropTypes.bool,
};

function ActiveIcon({ isActive }) {
    if (isActive) {
        return (
            <FontAwesomeIcon
                icon="fa-check"
                className="text-success text-center"
                title={_("Device is active")}
            />
        );
    }
    return (
        <FontAwesomeIcon
            icon="fa-times"
            className="text-danger"
            title={_("Device is inactive")}
        />
    );
}
