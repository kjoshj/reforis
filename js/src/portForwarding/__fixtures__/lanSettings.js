const lanSettingsFixture = {
    mode_managed: {
        dhcp: {
            clients: [
                {
                    active: true,
                    expires: 1739233767,
                    hostname: "MacBook",
                    ip: "192.168.2.101",
                    mac: "CA:21:77:1E:AE:72",
                    static: true,
                },
                {
                    active: false,
                    expires: 1739233392,
                    hostname: "iPhone",
                    ip: "192.168.2.102",
                    mac: "26:48:9E:51:41:29",
                    static: false,
                },
            ],
            enabled: true,
            lease_time: 12,
            limit: 150,
            start: "192.168.2.100",
        },
        netmask: "255.255.255.0",
        router_ip: "192.168.2.1",
    },
};

export default lanSettingsFixture;
