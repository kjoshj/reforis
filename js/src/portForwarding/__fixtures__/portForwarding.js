const portForwardingFixture = {
    rules: [
        {
            name: "Service",
            dest_ip: "192.168.2.101",
            dest_port: 8090,
            src_dport: 8090,
            enabled: true,
        },
    ],
};

export default portForwardingFixture;
