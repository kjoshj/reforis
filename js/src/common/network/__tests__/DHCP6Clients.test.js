/*
 * Copyright (C) 2019-2022 CZ.NIC z.s.p.o. (https://www.nic.cz/)
 *
 * This is free software, licensed under the GNU General Public License v3.
 * See /LICENSE for more information.
 */

import React from "react";
import { render } from "foris/testUtils/customTestRender";

import DHCP6Clients from "common/network/DHCP6Clients";
import { ipv6clients } from "./__fixtures__/clients";

describe("<DHCP6Clients/>", () => {
    it("Test with snapshot.", () => {
        const { container } = render(
            <DHCP6Clients ipv6clients={ipv6clients} />
        );
        expect(container).toMatchSnapshot();
    });
});
