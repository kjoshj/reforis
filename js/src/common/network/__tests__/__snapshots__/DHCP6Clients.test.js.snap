// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DHCP6Clients/> Test with snapshot. 1`] = `
<div>
  <div
    class="card p-4 col-sm-12 col-lg-12 p-0 mb-4"
  >
    <h2>
      IPv6 DHCP Client List
    </h2>
    <p>
      This list contains all devices that are connected to the network through wired or wireless connections using IPv6.
    </p>
    <div
      class="table-responsive"
    >
      <table
        class="table table-hover text-nowrap"
      >
        <thead
          class="thead-light"
        >
          <tr
            role="row"
          >
            <th
              colspan="1"
            >
              <button
                class="btn btn-link text-decoration-none text-reset fw-bold p-0 d-flex align-items-center
                                                    d-flex align-items-center
                                                "
                title="Sort ascending"
                type="button"
              >
                Hostname
              </button>
            </th>
            <th
              colspan="1"
            >
              <button
                class="btn btn-link text-decoration-none text-reset fw-bold p-0 d-flex align-items-center
                                                    d-flex align-items-center
                                                "
                title="Sort ascending"
                type="button"
              >
                IPv6 Address
              </button>
            </th>
            <th
              colspan="1"
            >
              <button
                class="btn btn-link text-decoration-none text-reset fw-bold p-0 d-flex align-items-center
                                                    d-flex align-items-center
                                                "
                title="Sort ascending"
                type="button"
              >
                DUID
              </button>
            </th>
            <th
              colspan="1"
            >
              <button
                class="btn btn-link text-decoration-none text-reset fw-bold p-0 d-flex align-items-center
                                                    d-flex align-items-center
                                                "
                title="Sort ascending"
                type="button"
              >
                Expires
              </button>
            </th>
            <th
              class="d-flex justify-content-center"
              colspan="1"
            >
              <button
                class="btn btn-link text-decoration-none text-reset fw-bold p-0 d-flex align-items-center
                                                    d-flex align-items-center
                                                "
                title="Sort descending"
                type="button"
              >
                Active
              </button>
            </th>
          </tr>
        </thead>
        <tbody>
          <tr
            class="align-middle"
          >
            <td>
              first
            </td>
            <td>
              2002:1487:fede:6110
            </td>
            <td>
              00049367d8183c4e45a4b834f5a92f95d3b9
            </td>
            <td>
              2018-10-12 13:16
            </td>
            <td
              class="text-center"
            >
              <i
                class="fa"
              />
            </td>
          </tr>
          <tr
            class="align-middle"
          >
            <td>
              second
            </td>
            <td>
              2002:1487:fefd:6130
            </td>
            <td>
              00049367d8183c4c43a4b834f5a92f95d3b9
            </td>
            <td>
              2018-10-14 20:49
            </td>
            <td
              class="text-center"
            >
              <i
                class="fa"
              />
            </td>
          </tr>
        </tbody>
      </table>
      <nav
        aria-label="Pagination navigation bar"
        class="d-flex gap-2 justify-content-start align-items-center mx-2 mb-1 text-nowrap"
      >
        <ul
          class="pagination pagination-sm mb-0"
        >
          <li
            class="page-item disabled"
            style="cursor: not-allowed;"
          >
            <button
              aria-label="First page"
              class="page-link"
              disabled=""
              type="button"
            >
              <i
                class="fa"
              />
            </button>
          </li>
          <li
            class="page-item disabled"
            style="cursor: not-allowed;"
          >
            <button
              aria-label="Previous page"
              class="page-link"
              disabled=""
              type="button"
            >
              <i
                class="fa"
              />
            </button>
          </li>
          <li
            class="page-item disabled"
            style="cursor: not-allowed;"
          >
            <button
              aria-label="Next page"
              class="page-link"
              disabled=""
              type="button"
            >
              <i
                class="fa"
              />
            </button>
          </li>
          <li
            class="page-item disabled"
            style="cursor: not-allowed;"
          >
            <button
              aria-label="Last page"
              class="page-link"
              disabled=""
              type="button"
            >
              <i
                class="fa"
              />
            </button>
          </li>
        </ul>
        <span>
          Page
           
          <span
            class="fw-bold"
          >
            1
             
            of
             
            1
          </span>
        </span>
        <div
          class="vr mx-1 align-self-center"
          style="height: 1.5rem;"
        />
        <span>
          Rows per page:
        </span>
        <select
          aria-label="Select rows per page"
          class="form-select form-select-sm w-auto"
        >
          <option
            value="5"
          >
            5
          </option>
          <option
            value="10"
          >
            10
          </option>
          <option
            value="25"
          >
            25
          </option>
          <option
            value="2"
          >
            All
          </option>
        </select>
      </nav>
    </div>
  </div>
</div>
`;
